pandas-stubs/core/indexes/base.pyi

```diff
@@ -195,15 +187,15 @@ class Index(IndexOpsMixin, PandasObject):
     def __ror__(self, other: Never) -> Never: ...
     def __xor__(self, other: Never) -> Never: ...
     def __rxor__(self, other: Never) -> Never: ...
-    def __neg__(self: IndexT) -> IndexT: ...
+    def __neg__(self) -> Self: ...
     def __nonzero__(self) -> None: ...
     __bool__ = ...
     def union(self, other: list[HashableT] | Index, sort=...) -> Index: ...
-    def intersection(self, other: list[T1] | Index, sort: bool = ...) -> Index: ...
-    def difference(self, other: list | Index, sort: bool | None = None) -> Index: ...
+    def intersection(self, other: list[S1] | Self, sort: bool = ...) -> Self: ...
+    def difference(self, other: list | Index, sort: bool | None = None) -> Self: ...
     def symmetric_difference(
-        self, other: list[T1] | Index, result_name=..., sort=...
```

Comment:
FYI - the `T1` references are from the original Microsoft stubs where `Index` was generic, but I couldn't ever get that to pass `mypy` (about 18 months ago).  So I just removed it, but probably missed the various references to `T1`

Reply:
I think I shouldn't use S1 here - would probably prevent comparing int with float. Should maybe use Scalar/Any (or wait for someone raising an issue).